rotkehlchen/db/dbhandler.py

```diff
@@ -3286,31 +3286,70 @@ def get_web3_nodes(self, only_active: bool = False) -> List[WeightedNode]:
         """
         with self.conn.read_ctx() as cursor:
             if only_active:
-                cursor.execute('SELECT name, address, owned, weight, active FROM web3_nodes WHERE active=TRUE AND CAST(weight as decimal) != 0;')  # noqa: E501
+                cursor.execute('SELECT identifier, name, endpoint, owned, weight, active FROM web3_nodes WHERE active=TRUE AND CAST(weight as decimal) != 0;')  # noqa: E501
             else:
-                cursor.execute('SELECT name, address, owned, weight, active FROM web3_nodes;')
+                cursor.execute(
+                    'SELECT identifier, name, endpoint, owned, weight, active FROM web3_nodes;',
+                )
             return [
                 WeightedNode(
+                    identifier=entry[0],
                     node_info=NodeName(
-                        name=entry[0],
-                        endpoint=entry[1],
-                        owned=bool(entry[2]),
+                        name=entry[1],
+                        endpoint=entry[2],
+                        owned=bool(entry[3]),
                     ),
-                    weight=FVal(entry[3]),
-                    active=bool(entry[4]),
+                    weight=FVal(entry[4]),
+                    active=bool(entry[5]),
                 )
                 for entry in cursor
             ]
 
+    def _rebalance_web3_nodes_weights(
+            self,
+            write_cursor: 'DBCursor',
+            proportion_to_share: FVal,
+            exclude_identifier: Optional[int],
+    ) -> None:
+        """
+        Weights for nodes have to be in the 0-1 range. So after setting a node weight to X
```

Comment:
So after your explanation I get what this does but let's change it a bit to make it more readable.

```
Weights for nodes have to be in the range between 0 and 1. This function adjusts the weights of all other nodes to keep the proportions correct.

After setting a node weight to X, the `proportion_to_share` between all remaining nodes becomes `1 - X`.

exclude_identifier is the identifier of the node whose weight we add or edit. In case of deletion it's omitted and `None`is passed.
```